src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -118,15 +162,72 @@ else if (this.superType != null)
      * @param name  Path of the registry's {@link ResourceLocation}
      * @param sup   Supplier of the RegistryBuilder that is called to fill {@link #type} during the NewRegistry event
      * @return      A supplier of the {@link IForgeRegistry} created by the builder.
+     *
+     * @deprecated Use {@link #create(ResourceLocation, String)} and {@link #makeRegistry(Class, Supplier)} instead
      */
+    @Deprecated(forRemoval = true, since = "1.18.2")
     public Supplier<IForgeRegistry<T>> makeRegistry(final String name, final Supplier<RegistryBuilder<T>> sup) {
-        if (this.superType == null)
-            throw new IllegalStateException("Cannot create a registry without specifying a base type");
-        if (this.type != null || this.registryFactory != null)
-            throw new IllegalStateException("Cannot create a registry for a type that already exists");
+        return makeRegistry(new ResourceLocation(modid, name), this.superType, sup);
+    }
+
+    /**
+     * Only used for custom registries to fill the forge registry held in this DeferredRegister.
+     *
+     * Calls {@link RegistryBuilder#setName} and {@link RegistryBuilder#setType} automatically.
+     *
+     * @param base The base type to use in the created {@link IForgeRegistry}
+     * @param sup Supplier of a RegistryBuilder that initializes a {@link IForgeRegistry} during the {@link RegistryEvent.NewRegistry} event
+     * @return A supplier of the {@link IForgeRegistry} created by the builder.
+     * Will always return null until after the {@link RegistryEvent.NewRegistry} event fires.
+     */
+    public Supplier<IForgeRegistry<T>> makeRegistry(final Class<T> base, final Supplier<RegistryBuilder<T>> sup) {
+        return makeRegistry(this.registryName, base, sup);
+    }
+
+    /**
+     * Creates a tag key based on the current modid and provided path as the location and the registry name linked to this DeferredRegister.
+     *
+     * @see #createOptionalTagKey(String, Set)
+     */
+    @NotNull
+    public TagKey<T> createTagKey(@NotNull String path)
```

Comment:
I think instead of a string this might make more sense to be a `ResourceLocation`? While I can see why it may be useful to have this be based on the deferred register's modid, I feel like especially for custom registries (which this and `createOptionalTagKey` are more likely to be used for as you can't easily static initialize optional tags for registries that aren't static init otherwise), it would be overly annoying to have to create a deferred register in another mod's domain *purely* for purposes of creating tags in their domain.

For example if an addon wanted to add an optional tag that has defaults for one of Mekanism's registries (say gases), instead of just being able to specify they may want the tag in our domain or in some other domain for another addon they are interacting with. They would have to create a second deferred register using our namespace in order to be able to create tags in it, rather than just being able to specify in the tag key creation they want it to be in our domain.